tests/bot/exts/moderation/test_silence.py

```diff
@@ -491,3 +748,86 @@ async def test_preserved_other_overwrites(self):
                 del new_overwrite_dict['add_reactions']
 
                 self.assertDictEqual(prev_overwrite_dict, new_overwrite_dict)
+
+    @mock.patch.object(silence.Silence, "_unsilence", return_value=False)
+    @mock.patch.object(silence.Silence, "send_message")
+    async def test_unsilence_helper_fail(self, send_message, _):
```

Comment:
I guess this is a similar situation to `test_sent_to_correct_channel`. You could probably merge this into `test_sent_correct_message` too. What you're doing with the custom class and the patching seems overcomplicated anyway.

Reply:
@HassanAbouelela This still needs to be addressed. `test_sent_correct_message` takes care of all the message stuff, so this test should absolutely not make any assertions on `send_message`. It only needs to assert the role with which `overwrites_for` was called. It doesn't even need to mock the return values of `overwrites_for` because `test_sent_correct_message` already tests for that.

Reply:
Yeah, this test is quite redundant. I cleaned it up in aef979e6.